packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiError.kt

```diff
@@ -42,13 +46,32 @@ class FlutterApiError {
             if (error.localizedMessage != null) {
                 localizedError = error.localizedMessage
             }
-            errorMap["PLATFORM_EXCEPTIONS"] = mapOf(
+            return createErrorMap(localizedError, recoverySuggestion, error.toString())
+        }
+        private fun createErrorMap(localizedError: String, recoverySuggestion: String?, errorString: String? = ""): Map<String, Any> {
+            var errorMap = HashMap<String, Any>()
+
+            errorMap.put("PLATFORM_EXCEPTIONS", mapOf(
                     "platform" to "Android",
                     "localizedErrorMessage" to localizedError,
                     "recoverySuggestion" to recoverySuggestion,
-                    "errorString" to error.toString()
-            )
+                    "errorString" to errorString
+            ))
             return errorMap
```

Comment:
Is this needed or can we use the `createErrorMap()` method defined below?

Reply:
It's needed because the kotlin methods require an actual exception object.